### DISCUSSION TITLE
How can I convert the LightGNN codes for example into a GPU version?

### DISCUSSION THREAD

**Discussion by dyoflut** [RUCAIBox/RecBole#1126] [Q&A]
Now, I have my GCMC codes as follows:

config = Config('GCMC', RecBole_args.dataset_name)
    init_logger(config)
    logger = getLogger()
    model = RecBole_general_benchmarks.GCMC(config, train_data.dataset).to(config['device'])
    trainer = Trainer(config, model)
    logger.info(model)
    best_valid_score, best_valid_result = trainer.fit(train_data, valid_data)
    test_result = trainer.evaluate(test_data)
    Evaluation_Metrics_to_CSV(this_round, RecBole_args, 'GCMC', test_result)

Based on these, I would like to convert them into a GPU version, such as by means of CUDA?

However, I still found no guide on RecBole Doc. about this.

In view of that, could please someone directly do something on the above codes of GCMC for that goal? Appreciate it a lot!

**Comment by Wicknight** [comment]
Hello! The default configuration of RecBole is the GPU version. I've noticed here that you don't seem to use the yaml file when creating the config.You can create a yaml file and set use_gpu and gpu_id parameters to use GPU. The reference [doc](https://recbole.io/docs/user_guide/config/environment_settings.html) is here. And you can also use torch.cuda.is_available() method to check whether GPU is available.
